wee_slack.py

```diff
@@ -1634,12 +1662,21 @@ class SlackDMChannel(SlackChannel):
 
     def __init__(self, eventrouter, users, **kwargs):
         dmuser = kwargs["user"]
-        kwargs["name"] = users[dmuser].name
+        kwargs["name"] = users[dmuser].name if dmuser in users else dmuser
         super(SlackDMChannel, self).__init__(eventrouter, **kwargs)
         self.type = 'im'
         self.update_color()
         self.set_name(self.slack_name)
-        self.topic = create_user_status_string(users[dmuser].profile)
+        if dmuser in users:
+            self.topic = create_user_status_string(users[dmuser].profile)
+
+    def set_related_server(self, team):
+        super(SlackDMChannel, self).set_related_server(team)
+        # If the topic hasn't been set then the user is external and needs to
+        # be fetched
+        if not self.topic:
```

Comment:
It's a bit fragile and weird to rely on the topic not being set. Instead you can rather check if `self.user not in self.team.users`.